core/state/state_object.go

```diff
@@ -221,11 +222,20 @@ func (s *stateObject) GetCommittedState(db Database, key common.Hash) common.Has
 	}
 	// If the snapshot is unavailable or reading from it fails, load from the database.
 	if s.db.snap == nil || err != nil {
+		var tr Trie
 		if s.db.GetTrie().IsVerkle() {
-			panic("verkle trees use the snapshot")
+			tr = s.db.GetTrie()
```

Comment:
Now that #209 has been merged, `getTrie` should be able to work for verkle trees as well if I use the adapter pattern from #198.